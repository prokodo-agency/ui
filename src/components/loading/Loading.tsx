import { createIsland } from "@/helpers/createIsland"

import LoadingServer from "./Loading.server"

import type { LoadingProps } from "./Loading.model"

export const Loading = createIsland<LoadingProps>({
  name: "Loading",
  Server: LoadingServer,
  loadLazy: () => import("./Loading.lazy")
})

export default Loading
