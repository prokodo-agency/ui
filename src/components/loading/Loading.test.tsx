import { expect } from "@jest/globals"

import { render } from "@/tests"

import { Loading } from "./Loading"

describe("The Loading component", () => {
  it(`should render Loading`, async () => {
    const { container } = render(<Loading size="xl" />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
