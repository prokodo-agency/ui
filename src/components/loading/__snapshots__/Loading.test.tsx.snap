// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Loading component should render Loading 1`] = `
<div
  class="prokodo-Loading"
  style="display: flex; align-items: center; justify-content: center; gap: 16px;"
>
  <svg
    height="0"
    style="position: absolute;"
    width="0"
  >
    <defs>
      <lineargradient
        class="prokodo-Loading__gradient"
        id="loading-gradient"
        x1="0%"
        x2="0%"
        y1="0%"
        y2="100%"
      >
        <stop
          offset="0%"
        />
        <stop
          offset="50%"
        />
        <stop
          offset="100%"
        />
      </lineargradient>
    </defs>
  </svg>
  <div
    class="prokodo-Loading__animation prokodo-Loading__animation--size-xl"
    style="border-radius: 50%; border: 4px solid transparent; border-top: 4px solid; animation: spin 1s linear infinite;"
  />
</div>
`;
